lib/vsc/mympirun/mpi/mpi.py

```diff
@@ -26,114 +26,131 @@
 Base MPI class, all actual classes should inherit from this one
 
 @author: Stijn De Weirdt
+@author: Jeroen De Clerck
 """
 
 import os
+import random
 import re
-import socket
-import shutil
-import time
 import resource
+import shutil
+import socket
 import stat
-import subprocess
-import random
 import string
-
+import subprocess
+import time
 
 from IPy import IP
+
 from vsc.utils.fancylogger import getLogger
 from vsc.utils.missing import get_subclasses, nub
-from vsc.utils.run import run_simple, run_simple_noworries, run_to_file, run_async_to_stdout
-
-# Going to guess myself
+from vsc.utils.run import run_simple, run_to_file, run_async_to_stdout
 
 # part of the directory that contains the installed fakes
 INSTALLATION_SUBDIRECTORY_NAME = '(VSC-tools|(?:vsc-)?mympirun)'
 # the fake subdir to contain the fake mpirun symlink
 # also hardcoded in setup.py !
 FAKE_SUBDIRECTORY_NAME = 'fake'
 
+_logger = getLogger()
+
 
 def whatMPI(name):
     """
-    Return the scriptname and the MPI class
+    Return the path of the selected mpirun and its class
+
+    @param name: The name of the executable used to run mympirun (sys.argv[0])
+
+    @return: A triplet containing the following variables:
+      - The path to the executable used to run mympirun (should be the path to an mpirun implementation)
+      - The corresponding python class of the MPI variant
+      - The python classes of the supported MPI flavors (from the various .py files in mympirun/mpi)
     """
-    fullscriptname = os.path.abspath(name)
-    scriptname = os.path.basename(fullscriptname)
 
-    found_mpi = get_subclasses(MPI)
+    scriptname = os.path.basename(os.path.abspath(name))
+    supp_mpi_impl = get_subclasses(MPI)  # support MPI implementations, these were imported in __init__.py
```

Comment:
`support` -> `supported`, and you should import them here, not in the `__init__`